grakn-graph/src/main/java/ai/grakn/graph/internal/ResourceTypeImpl.java

```diff
@@ -68,6 +69,16 @@ private ResourceTypeImpl(ResourceTypeImpl resourceType){
     }
 
     /**
+     * This method is overridden so that we can check that the regex of the new super type (if it has a regex)
+     * can be applied to all the existing instances.
+     */
+    @Override
+    public ResourceType<D> superType(ResourceType<D> superType){
+        checkInstancesMatchRegex(superType.getLabel(), superType.getRegex());
+        return super.superType(superType);
+    }
```

Comment:
I suspect we might have a subtle bug (which existed before, to be fair):
Given we have three types `foo`, `bar`, `baz` with regexes and instances, plus `foo sub bar;`.
If we state `baz sub foo`, it will check that instances of `baz` satisfy the `foo` regex, but not `bar`.